src/http/Android.C

```diff
@@ -17,6 +19,11 @@ void preventRemoveOfSymbolsDuringLinking() {
 #ifdef ANDROID
   extern "C" {
     int main(int argc, const char** argv);
+    
+    void runMainAndCleanup(int argc, const char** argv) {
```

Comment:
```suggestion
    void runMainAndCleanup(int argc, std::unique_ptr<const char*[]> argv)
```

?

Reply:
This seems wrong: this code is taking steps to preserve the `char*` array, but the actual content inside that is in the `vector<string>` in the enclosing code and this is just letting that get nuked

Reply:
While I agree that's it much nicer to have explicit ownership, the provided code looks ok to me.

This file has such low impact anyway, so I think it's fine to just go ahead with this.

Reply:
>While I agree that's it much nicer to have explicit ownership, the provided code looks ok to me.

What keeps the pointed to data alive? It seems that the whole reason to do this on a separate thread is so that it can happen after main() returns, but after main() returns, the data the char*s point to here are dead.

Reply:
There's no guarantee indeed. I suspect the data is used soon enough as to not cause an issue.

This piece of code is barely used, and internally we discussed dropping this back-end altogether (although that isn't final), so it didn't seem worth the wait for a fixup, or perform that myself.

From a pure coding perspective, I agree that it's best practice, but I was overruled :wink:

Reply:
Why not just move the `vector<string>` to the other thread instead?